dinstaller-lib/src/settings.rs

```diff
@@ -0,0 +1,141 @@
+use crate::users::{UsersClient,FirstUser};
+use crate::storage::StorageClient;
+use std::{str::FromStr, error::Error};
+
+#[derive(Debug, Default)]
+pub struct Settings {
+    pub users: Option<UsersSettings>,
+    pub storage: Option<StorageSettings>
+}
+
+#[derive(Debug)]
+pub struct UsersSettings {
+    pub first_user: Option<FirstUser>
+}
+
+#[derive(Debug)]
+pub struct StorageSettings {
+
+}
+
+/// Settings storage
+///
+/// It is responsible for loading and storing the settings in the D-Bus service.
+pub struct Store<'a> {
+    storage_client: StorageClient<'a>,
+    users_client: UsersClient<'a>,
+}
+
+impl<'a> Store<'a> {
+    pub fn new() -> Result<Self, Box<dyn Error>> {
+        Ok(
+            Self {
+                storage_client: StorageClient::new(super::connection()?)?,
+                users_client: UsersClient::new(super::connection()?)?
+            }
+        )
+    }
+
+    /// Loads the installation settings from the D-Bus service
+    pub fn load(&self) -> Result<Settings, Box<dyn Error>> {
+        let mut settings = Settings::default();
+        settings.users = Some(UsersSettings {
+            first_user: Some(self.users_client.first_user()?)
```

Comment:
I worry that this load/store methods grows significantly. I think that Store should not hold clients and do all communication and instead deletegate it to specific structs like UserSettings...if you do not like having in Settings that loading, then I would create specific methods or structs like UsersLoader/Storer that just get shortly reference to client it need.

Reply:
Yes, I agree. We can do that later as I expect we add more logic to load data from D-Bus rather soon.